src/main/java/frc/robot/subsystems/endgame/EndgameIOSparkFlex.java

```diff
@@ -22,15 +38,59 @@ public EndgameIOSparkFlex() {
 
         leftEndgameMotor.getEncoder().setPositionConversionFactor(EndgameConstants.encoderToMeters);
         leftEndgameMotor.getEncoder().setPosition(0.0);
+
+        rightEndgameMotor.getPIDController().setP(EndgameConstants.climberkP);
+        rightEndgameMotor.getPIDController().setI(EndgameConstants.climberkI);
+        rightEndgameMotor.getPIDController().setD(EndgameConstants.climberkD);
+        rightEndgameMotor.getPIDController().setFF(EndgameConstants.climberkFFClimber);
     }
 
     @Override
-    public void setVolts(double volts) {
-        rightEndgameMotor.setVoltage(volts);
+    public void setOverrideMode(boolean override) {
+        this.override = override;
+    }
+
+    @Override
+    public void setOverrideVolts(double volts) {
+        overrideVolts = volts;
+    }
+
+    @Override
+    public void setPosition(double position) {
+        if (targetPosition != position) {
+            profileTimer.reset();
+            profileTimer.start();
+
+            targetPosition = position;
+
+            initialPosition = rightEndgameMotor.getEncoder().getPosition();
+            initialVelocity = rightEndgameMotor.getEncoder().getVelocity();
+
+            if (position < targetPosition) {
+                // Moving down, assume weight of robot is on the climber now
+                setFF(EndgameConstants.climberkFFRobot);
+            } else {
+                // Moving up, we only have to account for the weight of pushing the climber up
+                setFF(EndgameConstants.climberkFFClimber);
+            }
+        }
     }
 
     @Override
     public void updateInputs(EndgameIOInputs inputs) {
+        if (override) {
+            rightEndgameMotor.setVoltage(overrideVolts);
+        } else {
+            State trapezoidSetpoint =
+                    profile.calculate(
+                            profileTimer.get(),
+                            new State(initialPosition, initialVelocity),
+                            new State(targetPosition, 0.0));
+            rightEndgameMotor
+                    .getPIDController()
+                    .setReference(trapezoidSetpoint.position, ControlType.kPosition);
```

Comment:
For sanity can we add a clamp before this to ensure the trapezoidSetpoint.position is never greater than the maximum allowed position of the climber